File: java/java_base/src/com/thzhima/javabase/basetype/People.java
package com.thzhima.javabase.basetype;

public class People {
	public String name;
	public int age;
	
	public People(String name, int age){
		this.age = age;
		this.name = name;
	}

	@Override
	public String toString() {
		return "People [name=" + name + ", age=" + age + "]";
	}
	
}

File: py/fruit/fruitserver/db/people.py
class People():
    def __init__(self, name, age):
        self.name = name
        self.age = age

    # def __init__(self, name):
    #     self.name = name
    #     self.age = None

    def work(a):
        print("{} working now".format(a.name))

    ''' Python 没有方法重载， 同名方法会覆盖之前定义的方法。'''
    # def work(a, task):
    #     print("{}do {} working now".format(a.name, task))

    def __str__(self):
        if self.age:
            return '{name:%s, age:%d}' %(self.name, self.age)
        else:
            return '{name:%s}' % (self.name)

    def __repr__(self):
        return self.__str__()


# p = People('wang')
# print(p)

p2 = People("Xie", 20)
print(p2)
p2.work()




